### DISCUSSION TITLE
How to use Custom Shortcuts to create a second shortcut set (using repeat commands)

### DISCUSSION THREAD

**Discussion by mickimnet** [rxhanson/RectanglePro-Community#5] [Show and tell]
One of the things, I love most about `Rectangle` is the option to set `Repeated commands` to `cycle ½, ⅔ and ⅓ on half actions`, which lets me (repeatedly) press just one shortcut to move and size a window to three different sizes. That is one of the highest performance boost for me, because I can group my windows in a couple of different sizes very fast.

Most of the time I work with an external keyboard and monitor and use the numpad 1-9 keys for directional placement (and sizing of the windows). But when I'm working on the laptop directly, I still would like to use at least 4 of the 8 possible directions. Luckily `Rectangle Pro` came to the rescue with `Custom Shortcuts`.

Therefore I would like to share how to do that exactly:

1. The basic shortcut setup done in `Keyboard Shortcuts`
<img width="723" alt="Rectangle Pro-01-Keyboard Shortcuts" src="https://user-images.githubusercontent.com/2911922/149908305-51794c85-684e-4ea1-a70c-aa85f37e222f.png">

2. To replicate the `cycle` action (see above), you first need to setup the shortcut for ½ of the screen
<img width="904" alt="Rectangle Pro-02-Custom Shortcuts" src="https://user-images.githubusercontent.com/2911922/149908369-d888b5f8-aa0c-489f-956e-bfd8a7e04448.png">

3. Add a second action to the shortcut click on the `+` symbol
<img width="747" alt="Rectangle Pro-03-Custom Shortcuts" src="https://user-images.githubusercontent.com/2911922/149908404-de234c7c-78f9-4043-a894-0d31066170ab.png">

4. Not it get's a bit tricky, in order to mimic the default behaviour, enter `0.6666`—otherwise the sizing won't fit with the standard shortcut
<img width="904" alt="Rectangle Pro-04-Custom Shortcuts" src="https://user-images.githubusercontent.com/2911922/149908545-4fa7d37d-ad3f-4337-b270-a7eb0a5e86e3.png">

5. To add the third action you need to click on the `+` symbol of the second action
<img width="747" alt="Rectangle Pro-05-Custom Shortcuts" src="https://user-images.githubusercontent.com/2911922/149908636-3262855d-88bc-4129-aa7a-4bc85ec3048e.png">

6.  As with the second action you need to have 4 decimal places: `0.3333`
<img width="904" alt="Rectangle Pro-06-Custom Shortcuts" src="https://user-images.githubusercontent.com/2911922/149908672-d6e9ac94-6001-417e-bd76-a680444f97bc.png">

Rince and repeat for all the shortcuts you like to use.
<img width="747" alt="Rectangle Pro-07-Custom Shortcuts" src="https://user-images.githubusercontent.com/2911922/149908724-d5a04021-2c19-479f-aab0-d516284a989d.png">

And @rxhanson thank you very much for creating [Rectangle](https://rectangleapp.com) and especially [Rectangle Pro](https://rectangleapp.com/pro) in the first place!

Take care
Mick

**Comment by rxhanson** [comment]
Thanks! You can also use 2/3 and 1/3 if you don't want to type 0.6666 or 0.3333.

**Reply by mickimnet** [reply]
Thank you! That's very good to know, as x/3 is more precise than an decimal place number!

**Comment by davidbernard04** [comment]
@mickimnet OMG thank you very much! I was heading here with the purpose to ask if it was possible to do exactly that, and bam!! here is a complete guide written 6 days ago. :D I almost missed it though, since I was looking for the "repeat commands" words, so maybe if you add this in your title it will be more obvious. 

For me too it's a great performance boost, especially since you have less shortcuts to memorize. :)

**Comment by agfirewire** [comment]
@mickimnet Thank you for this post, super helpful

**Comment by kennonb** [comment]
Thanks for this @mickimnet! 🎉 I was curious if this was possible, and your walkthrough helped me get this setup on my system. 😃

**Comment by 5knnbdwm** [comment]
@rxhanson is there a way to have the same key combi do something different on different monitors?

**Reply by rxhanson** [reply]
Not yet, but it's on my backlog.

**Reply by mickimnet** [reply]
That would be great, because I have two monitors, one in landscape and one in portrait mode, where specific custom shortcuts should behave differently.

**Comment by peterinjapan** [comment]
Just wanted to say this worked well for me, thanks very much! Now I can stop obsessing over how to automate my windows and get back to work!
